src/main/java/uk/gov/justice/digital/delius/data/api/NewCourtDto.java

```diff
@@ -0,0 +1,36 @@
+package uk.gov.justice.digital.delius.data.api;
+
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+
+import javax.validation.constraints.NotBlank;
+
```

Comment:
Note the use of `record` - see main comment where I don't see enough benefits to use this -  the 2 big disadvantages you will see in the unit tests

1. No ability to copy and existing instance and copy (e.g like kotlin `copy` or lombok `toBuilder`
2. No ability to name constructor parameters - so you end up with a huge constructor where parameter ordering matters (unlike kotlin named parameters or a builder pattern)